src/logger/logger.class.ts

```diff
@@ -1,20 +1,20 @@
 import { Level } from './level'
 import { Format, FormatOptions } from './format'
-import { createVault, VaultOptions, IVault, Vault } from './vault'
-import { ILogger } from './logger.interface'
+import { createVault, VaultOptions, Vault, VaultKind } from './vault'
+import { Logger } from './logger.interface'
 import { LogContext } from './log-context.class'
 
-export type LoggerOptions = {
+export type LoggerBaseOptions = {
   level: Level
   format: FormatOptions
   vault: VaultOptions | VaultOptions[]
 }
 
-export class Logger implements ILogger {
-  vaults: IVault[]
+export class LoggerBase implements Logger {
```

Comment:
Might be nothing, but is `LoggerBase` ever instantiated directly or only ever through inheriting class?
If it isn't instantiated directly, and never should, isn't it missing the `abstract` keyword, to enfore that?

Reply:
Not in the code, but it is exported to be used in the projects, where the logger library is included.